### ISSUE TITLE
Do not manage `$AWS_DEFAULT_REGION`

### ISSUE THREAD

**Issue by synfinatic** [synfinatic/aws-sso-cli#149]
This ticket is basically my thoughts on why I need to remove the `--region` flag and `DefaultRegion` config variables.

Basically managing this variable IMHO violates one of the basic rules of any app: no surprises/side effects.  People use `aws-sso` to select a role in an account.  Accounts often have resources launched in many regions.  In modern large scale application design, there is no correct "primary" region.  This isn't a timezone where everything is indexed off of GMT+0.

So the fundamental assumption that when I switch between roles in an account or between accounts that I should also want to change my default region seems flawed.  It also has a number of bad side effects:

 * Users may need to switch roles and find that they are running commands in the wrong region because they forget they had changed a default setting for that one account 6 months ago. 
 * `aws-sso` does not enforce users to be consistent- you can set the DefaultRegion at any level (or no level at all) and you may end up with some roles having a default region and others not having one.
 * `aws-sso` setting the region will not always honor the region manually set via `$AWS_DEFAULT_REGION`.
 * Supporting this functionality increases code complexity
 * Needing to deal with a number of corner cases and having to explain the complicated business logic to users is hard/impossible.   _Different users expect different behavior!_
 * There are existing bugs today and fixing them will likely only increase code complexity

Specifically we end up with really hard questions like:

If the user uses `aws-sso eval --region xxxx` where xxx overrides the current `$AWS_DEFAULT_REGION`, what should `aws-sso eval --clear` do?  Do we leave the variable alone or do we try to reset it?  How do we know what the old value was?

The argument for keeping the feature is that it we can make it fully opt-in (need to remove the defaults like: https://github.com/synfinatic/aws-sso-cli/pull/148/files#L76) and then add another variable like `$AWS_SSO_OLD_REGION`, but then you have to worry about users switching between multiple roles so it really needs to be an array...  ugh. 

Basically it seems like a lot of complexity, for little gain (again, this whole feature seems counter to how modern well designed services are deployed & managed in AWS).  Of course there are tons of orgs with badly designed & deployed services in AWS, but I'm still not sure what the user story is?

**Comment by synfinatic** [comment]
The counter to the above argument is that this feature was intended to emulate how AWS allows you to set a default region and/or per-profile region in the `~/.aws/config` file.  The challenge here of course is that if you are using the [aws cli](https://aws.amazon.com/cli/) with`aws-sso`, then the `aws` command will honor the `[default]` region, but not any `[profile]` regions because there are no 'profiles'.  (Although I had considered this in #15)

So there is precedent for this region feature- although perhaps my current implementation has some holes that need to be cleaned up.

I think what needs to happen, is I need to have a clear document (dot flowchart?) which explains exactly how this logic should work and can unit test.

**Comment by synfinatic** [comment]
So we definitely should match the AWS `~/.aws/config` profile configuration options of controlling the default role on a per-profile & set a default.  So this indicates needing to support the `DefaultRegion` in the config.

Since the `$AWS_DEFAULT_REGION` should always take priority, we should _never_ change that value.  But that is the only way to set it from the config (since no AWS API library knows to look in our config file).

That means the open question is should we support the `--region` flag.  The answer seems to be "no", because users should just set `$AWS_DEFAULT_REGION` like normal.

Hence this ticket as described is invalid.  closing as invalid.  Will open a new one.
